hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsOutputStream.java

```diff
@@ -345,6 +345,7 @@ private void uploadBlockAsync(DataBlocks.DataBlock blockToUpload,
             return null;
           } finally {
             IOUtils.close(blockUploadData);
+            blockToUpload.close();
```

Comment:
right, because the line above calls BlockUploadData.close(), a lot of the cleanup should take place already; that .startUpload() call on L315 sets the blockToUpload.buffer ref to null, so that reference doesn't retain a hold on the bytebuffer.

this is why we haven't seen problems yet like OOM/disk storage...cleanup was happening.

can you review the code to make sure the sequence of L348 always does the right thing and not fail due to some attempted double delete of the file...

Reply:
It is safe in case of file delete, as java.io.File will return false in case file is not there.